### PR TITLE
[RISC-V, LoongArch64] Fix assert(!regArgMaskLive) in genFnPrologCalleeRegArgs

### PR DESCRIPTION
**Pull request by t-mustafin**
Fixes one of asserts in System.Text.RegularExpressions.Tests.

Part of https://github.com/dotnet/runtime/issues/84834
cc @alpencolt @gbalykov @ashaurtaev @clamp03 @tomeksowi

### CODE REVIEW

**Comment by t-mustafin** [comment]
cc @shushanhf

**Review by clamp03 — APPROVED** [review]

**Comment by shushanhf** [comment]
> cc @shushanhf

Thanks
LGTM

**Review by alpencolt — APPROVED** [review]

File: src/coreclr/jit/codegenriscv64.cpp [hunk 1]
```diff
@@ -8066,6 +8067,7 @@ void CodeGen::genFnPrologCalleeRegArgs()
                                                             (regNumber)regArg[i], (regNumber)regArg[i]);
                                 regArgNum--;
                                 regArgMaskLive &= ~genRegMask((regNumber)regArg[j]);
+                                regArg[j] = 0;
```
[the remark this answers is not in the document]
Comment (shushanhf) [reply src/coreclr/jit/codegenriscv64.cpp:8070]: In fact, this `0`  is not the `REG_R0`, the `0`  is just only an initial value which represents empty.
Only the argument registers is valid liking a0-a7 or  f0-f7.

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/codegenriscv64.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (t-mustafin) [reply src/coreclr/jit/codegenriscv64.cpp:8070]: 0 or `REG_R0` value here means register number `j` is freed of values and some new value can be moved into it.

**Pull request merged**